### PR TITLE
Basic typescript and tools setup

### PR DESCRIPTION
**Pull request by sachitsac**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "env": {
+    "node": true,
+    "es2020": true
+  },
+  "extends": ["@mscharley"],
+  "parser": "@typescript-eslint/parser",
```
Comment (mscharley) [inline .eslintrc.json]: Setting parser explicitly isn't necessary with my ruleset and can be actively harmful in some cases.

**Comment by sachitsac** [comment]
Fair enough. I will remove it. I had it before I put the ruleset.

**Comment by mscharley** [comment]
I figured out the export issue too. `src/math.ts` doesn't exist, only `src/Math.ts` does.

**Comment by sachitsac** [comment]
> I figured out the export issue too. `src/math.ts` doesn't exist, only `src/Math.ts` does.

I'll be damed. Yeah, that did it. Thanks @mscharley. Sometimes the most basic things are the ones that catches us the most.

**Comment by sachitsac** [comment]
@mscharley 

So i am happy with where this is at now. 

### One question: 

- Why do you use [Babel](https://babeljs.io/) instead of tsc in your node presets ? What are the advantages of that over just using typescript CLI ?

**Comment by mscharley** [comment]
@sachitsac Where do I use Babel? I can't see any reference to babel in the presets at all.

**Comment by sachitsac** [comment]
You have a [babel config](https://github.com/mscharley/node-presets/blob/master/commands/bootstrap/babel.config.js) but i cannot see it being used though. But was wondering why you had it.

**Comment by mscharley** [comment]
Ah, yeah. That's not in the presets themselves. I can't remember why that's in there offhand to be honest. It'll be to do with Jest, but it may even just be a holdover from when I was using Jest with ReasonML which does need something like Babel to process the JS files into something that node could deal with at the time. At any rate, babel is being used by jest, i'm just not sure if it's necessary in a TS project.

**Comment by mscharley** [comment]
It's probably just so that Jest can run the TS tests directly without needing a compile run.

**Comment by sachitsac** [comment]
Ah ok. All good ... i was just curious why it was being used. Thanks for the explaining. Next up, simple graphql server. Will send you the PR if you have time to review it. Thanks @mscharley

**Pull request merged**